xm-commons-config/src/main/java/com/icthh/xm/commons/config/client/api/AbstractConfigService.java

```diff
@@ -28,12 +28,17 @@ public void addConfigurationChangedListener(ConfigurationChangedListener configu
     @Override
     public void updateConfigurations(String commit, Collection<String> paths) {
         Map<String, Configuration> configurationsMap = getConfigurationMap(commit, paths);
-        paths.forEach(path -> notifyUpdated(configurationsMap
-            .getOrDefault(path, new Configuration(path, null))));
+        paths.forEach(path -> notifyUpdated(getNonNullConfiguration(configurationsMap, path)));
     }
 
     protected void notifyUpdated(Configuration configuration) {
-        log.debug("Notify configuration changed [{}]", configuration.getPath());
+        log.debug("Notify configuration changed [{}]", configuration != null ? configuration.getPath() : null);
```

Comment:
I think need better handle case when configuration == null.

Because in this case we will have potential npe in nested class

configurationListener.onConfigurationChanged(configuration)